Source/Core/InputCommon/ControllerInterface/Win32/Win32.cpp

```diff
@@ -125,10 +132,14 @@ void ciface::Win32::PopulateDevices(void* hwnd)
   if (s_thread.joinable())
   {
     s_hwnd = static_cast<HWND>(hwnd);
-    s_done_populating.Reset();
+    s_first_pupulate_devices_asked = true;
+    s_received_device_change_event.Reset();
+    // Do this forced devices refresh in the messaging thread so it won't cause any race conditions
     PostMessage(s_message_window, WM_INPUT_DEVICE_CHANGE, 0, 0);
-    if (!s_done_populating.WaitFor(std::chrono::seconds(10)))
-      ERROR_LOG_FMT(CONTROLLERINTERFACE, "win32 timed out when trying to populate devices");
+    std::thread([] {
+      if (!s_received_device_change_event.WaitFor(std::chrono::microseconds(10)))
```

Comment:
10µs seems very, very short for Windows.

Reply:
Good spot. This was a mistake, it used to be 10 seconds, I didn't mean to change it by that much.